parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -19,6 +20,21 @@ source {{GetCSEHelpersScriptFilepath}}
 wait_for_file 3600 1 {{GetCSEInstallScriptFilepath}} || exit $ERR_FILE_WATCH_TIMEOUT
 source {{GetCSEInstallScriptFilepath}}
 
+{{- if HasTelemetryEnabled }}
+installAPMZ
+
+function exitAndFlush {
+  if [[ $? != "0" ]]; then
+    trace_err "cse-exit" code=$?
+  else
+    trace_info "cse-exit" code=$?
+  fi
+  apmz batch -f ${TMP_APMZ_BATCH_FILE} --api-key {{GetApplicationInsightsTelemetryKey}}
```

Comment:
does this ever fail intermittently? Worth retrying?

Reply:
retry is handled within apmz